### PR TITLE
Fix display of Connection menu

### PR DESCRIPTION
**Pull request by mmaret**
UDP was not displayed in the menu but using down key show the details
associated to UDP

### CODE REVIEW

**Comment by AJenbo** [comment]
Probably we should create an enum for  this instead so that we don't have to reorder the  menu items.

**Comment by mmaret** [comment]
I'm not very happy with the implementation.
Having #ifdef around Enum item is not very nice and we still need the enum to be in sync with the connection dialog.
The last one could have been avoided by using
``` 
UiListItem SELCONN_DIALOG_ITEMS[] = {
[SELCONN_LOOPBACK] = { "Loopback", SELCONN_LOOPBACK }
...}
```
But even gcc 9 complains about it with `sorry, unimplemented: non-trivial designated initializers not supported` so it would not be very portable.

Let me know what you think about that

File: SourceX/DiabloUI/selconn.cpp [hunk 1]
```diff
@@ -67,18 +78,22 @@ void selconn_Focus(int value)
 {
 	int players = MAX_PLRS;
 	switch (value) {
-	case 0:
+	case SELCONN_LOOPBACK:
```
Comment (AJenbo) [inline SourceX/DiabloUI/selconn.cpp]: Any reason this was moved to the top in both switches?

**Comment by AJenbo** [comment]
Thank you for the clean up

**Pull request merged**